### DISCUSSION TITLE
allow custom SMS OTP messages from client

### DISCUSSION THREAD

**Discussion by anngbaum** [supabase/supabase#10873] [Feature Requests]
We have an active iOS and Android app. Both use SMS OTP for verification.

On iOS, the auto-fill with the OTP works very nicely out of the box.

On Android, however, we have received user complaints that login is clunky. With Google's [current offering](https://developers.google.com/identity/sms-retriever/verify#1_construct_a_verification_message), my understanding is that we can offer similar functionality, but it requires adding an ugly hash to the end of the message:

![Screen Shot 2022-12-12 at 10 36 24 AM](https://user-images.githubusercontent.com/29735378/207086831-f11380d5-c825-446d-83c5-997debb274a7.png)

One way I thought this could work better is if the client could pass an optional parameter up to the `verifyOTP` endpoint like `additional_info`, and have that be appended to the message in the same way as `{{ .Code }}`. That way if the client knows it's an Android device, we could append that hash. So from Android, we would simply run:

`let { session, error } = await supabase.auth.verifyOTP({
  phone: '+13334445555',
  token: '123456',
  type: 'sms',
  additional_info: 'FA+9qCX9VSu',
})`

and the SMS language could be:
`Welcome to Spillt. Your code is {{ .Code }}  {{ .AdditionalInfo}}`

Alternatively, if this is a service many mobile users want to support (and I would guess they do!) there could just be an explicit "Google Hash ID" parameter in the Twilio settings, and let the client specify device type as a more structured parameter.
![Screen Shot 2022-12-12 at 10 33 37 AM](https://user-images.githubusercontent.com/29735378/207088154-fe7603c6-d6c9-4d8e-8a5c-3135335daf03.png)

These were my ideas (though of course any solution would be much appreciated so that we don't have to add the hash unnecessarily to iOS messages!)

**Comment by marcus-sa** [comment]
Is this still not supported?

**Reply by anngbaum** [reply]
no, not to my knowledge

**Reply by marcus-sa** [reply]
@anngbaum damn, no IAM has proper support for SMS OTP. All I want is SMS OTP and WebAuthn haha.
Looks like I'm gonna have to roll my own.

**Comment by A-Ghattas** [comment]
Hi, I just want to check on the status of this feature because I want it too.

**Reply by marcus-sa** [reply]
@A-Ghattas don't expect anything 
from Supabase. They keep releasing new products instead of useful features lol
